### DISCUSSION TITLE
Simple way to create a new app?

### DISCUSSION THREAD

**Discussion by hazytea** [nasa/cFS#765] [Q&A]
With the new files in /config and data sheets in the main branch, is there an accepted way to create a new app which doesn't involve copy/pasting sample_app and reformatting/replacing text within it to create the new app?  Writing a script to do it feels awkward and I'd spend more time chasing edge cases than I would doing it manually.  Has someone created something to do this, or is there a more elegant solution?  Thanks.

**Comment by mbacch** [comment]
I am interested in this too. Has anyone looked at how NASA's F' does this? I've only performed the beginner HelloWorld tutorial but found it nice to run one command to generate a new component: 
`fprime-util new --component`
Link to that tutorial is [here](https://fprime-community.github.io/fprime-tutorial-hello-world/docs/HelloWorld.html). Something like this would be useful in the CFS ecosystem if it already doesn't exist

**Comment by SterlingPeet** [comment]
I wrote a tool to stamp out an App looks approximately like the sample app, here: https://github.com/SterlingPeet/cookiecutter-cfs-app

It worked well enough to solve some compilation problems I was having across a handful of apps by applying the template.  Maybe it will work for you too.  Its not a highly polished or supported tool, so YMMV.

Full Disclosure: I also wrote the the first F Prime component template that was ultimately the baseline for ``fprime-util new``.

**Reply by SterlingPeet** [reply]
FYI, this was written for compatibility with Draco RC4.

**Reply by rkt34cornell** [reply]
Does this generate a config folder as well? I keep getting an error that says - cannot open shared object file: No such file or directory

The .so file was generated and is in the /cf directory but cannot be recognized for some reason

**Reply by SterlingPeet** [reply]
As it is, the template does not modify your startup script.  You would need to manually add the new app to the startup script in order for cFE to load the new app at startup.  I think you can also dynamically load the app once cFE is running via command, but I would recommend adding it to the startup script.
